crates/sqlbuiltins/src/functions/table/object_store.rs

```diff
@@ -697,32 +678,94 @@ impl TableFunc for CloudUpload {
             storage.db_id(),
             file_name
         ));
-        let base_url = storage.root_url.to_string();
-        let base_url = ObjectStoreUrl::parse(base_url)?;
 
         // No globbing: just retrieve meta for the requested file
-        let meta: ObjectMeta = match store.head(&path).await {
+        let meta: ObjectMeta = match storage.store.head(&path).await {
             Ok(meta) => meta,
             Err(_) => {
                 return Err(ExtensionError::String(format!(
                     "file not found: {}",
-                    file_name
+                    file_name,
                 )))
             }
         };
-        let objects: Vec<ObjectMeta> = vec![meta]; // needed for ObjStoreTableProvider impl
-
-        let session_state = ctx.get_session_state();
-        let arrow_schema = file_format
-            .infer_schema(&session_state, &store.inner, &objects)
-            .await?;
-
-        Ok(Arc::new(ObjStoreTableProvider::new(
-            store.inner.clone(),
-            arrow_schema,
-            base_url,
-            objects,
-            file_format,
-        )))
+
+        let ext = Path::new(&file_name)
+            .extension()
+            .ok_or_else(|| {
+                ExtensionError::String(
+                    "missing file extension, supported: [.csv, .json, .parquet]".to_string(),
+                )
+            })?
+            .to_str()
+            .ok_or_else(|| {
+                ExtensionError::String(format!("unsupported file extension: {file_name}"))
+            })?
+            .to_lowercase();
+
+        match ext.as_str() {
+            "csv" => {
+                let file_format =
+                    Arc::new(CsvFormat::default().with_schema_infer_max_rec(Some(20480)));
+                object_store_table_from_file_format(ctx, file_format, &storage, meta).await
+            }
+            "ndjson" | "jsonl" => {
```

Comment:
ok, so it will now support `ndjson` and `jsonl` in addition to excel and bson from the title - is that correct?

Reply:
So `"json"` has been renamed to `"ndjson"` and `"jsonl"` (`read_ndjson`).

`"json"` corresponds to `read_json`.

And BSON, Excel support has been added.

Reply:
Yep makes sense. I tested `jsonl` and `xlsx` on Cloud QA.

If you have a BSON file handy to test, that would be helpful.